feat: document new options (no domains yet) (#1777)
biomejs/website: src/content/docs/reference/configuration.mdx

- style: Rules enforcing a consistent and idiomatic way of writing your code.
- suspicious: Rules that detect code that is likely to be incorrect or useless.

Each group can accept, as a value, a string that represents the severity or an object where each rule can be configured.

When passing the severity, you can control the severity emitted by all the rules that belong to a group.
For example, you can configure the `a11y` group to emit information diagnostics:

```json title="biome.json"
{
  "linter": {
    "rules": {
      "a11y": "info"
    }
  }
}
```

Here's the accepted values:

> - `"on"`: each rule that belongs to the group will emit a diagnostic with the default severity of the rule. Refer to the documentation of the rule, or use the `explain` command:

Author: I am not very fond of this wording, it doesn't feel very clear. If you have a better suggestion, please offer me one :)

Author: It is? That's great! It's just that it took me a while to write it, so I wasn't sure about the final product

> Default: `"preserve"`

### `json.formatter.expand`

Whether to expand arrays and objects literals on multiple lines.

> - `"followSource"`: arrays and objects literals are formatted by following Prettier heuristics

Reviewer: Might be better to explain the hueristic. I wouldn't expect everyone to know the Prettier behaviour.

```suggestion
- `"followSource"`: arrays and objects literals are formatted on multiple lines if they already were on multiple lines, or if they don't fit on a single line.
```

> Default: `"double"`

### `css.linter.enabled`

Enables Biome's linter for CSS files.

> Default: `true`

```json title="biome.json"
{
  "css": {
    "linter": {
      "enabled": false
    }
  }
}
```

### `css.assist.enabled`

Enables Biome's assist for CSS files.

> Default: `true`

```json title="biome.json"
{
  "css": {
    "assist": {
      "enabled": false
    }
  }
}
```

## `graphql`

Options applied to the GraphQL files.

### `graphql.formatter.enabled`

Enables Biome's formatter for GraphQL files.

> Default: `false`

### `graphql.formatter.indentStyle`

The style of the indentation for GraphQL files. It can be `"tab"` or `"space"`.

> Default: `"tab"`

### `graphql.formatter.indentWidth`

How big the indentation should be for GraphQL files.

> Default: `2`

### `graphql.formatter.lineEnding`

The type of line ending for GraphQL files.
- `"lf"`, Line Feed only (`\n`), common on Linux and macOS as well as inside git repos;
- `"crlf"`, Carriage Return + Line Feed characters (`\r\n`), common on Windows;
- `"cr"`, Carriage Return character only (`\r`), used very rarely.

> Default: `"lf"`

### `graphql.formatter.lineWidth`

> How many characters can be written on a single line in √ files.

Reviewer: ```suggestion
The amount of characters that can be written on a single line in GraphQL files.
```

Author: This is a copy paste of all other descriptions. I will update all of them

> Default: `80`

### `graphql.formatter.quoteStyle`

The type of quote used when representing string literals. It can be `"single"` or `"double"`.

> Default: `"double"`

### `graphql.linter.enabled`

Enables Biome's linter for GraphQL files.

> Default: `true`

### `graphql.assist.enabled`

Enables Biome's assist for GraphQL files.

> Default: `true`

## `grit`

Options applied to the Grit files.

### `grit.formatter.enabled`

Enables Biome's formatter for Grit files.

> Default: `false`

### `grit.formatter.indentStyle`

The style of the indentation for Grit files. It can be `"tab"` or `"space"`.

> Default: `"tab"`

### `grit.formatter.indentWidth`

How big the indentation should be for Grit files.

> Default: `2`

### `grit.formatter.lineEnding`

The type of line ending for Grit files.
- `"lf"`, Line Feed only (`\n`), common on Linux and macOS as well as inside git repos;
- `"crlf"`, Carriage Return + Line Feed characters (`\r\n`), common on Windows;
- `"cr"`, Carriage Return character only (`\r`), used very rarely.

> Default: `"lf"`

### `grit.formatter.lineWidth`

> How many characters can be written on a single line in √ files.

Reviewer 2: nit
```suggestion
How many characters can be written on a single line in Grit files.
```
